numpy/f2py/__init__.py

```diff
@@ -32,7 +36,7 @@ def compile(source,
         Fortran source of module / subroutine to compile
     modulename : str, optional
         The name of the compiled python module
-    extra_args : str, optional
+    extra_args : str or list, optional
```

Comment:
Should I add a `versionchanged` directive?

Reply:
Sure, why not

Reply:
added below